### PR TITLE
Storybook: Add Color Palette Component

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Adds Color Palette component to storybook.

## How has this been tested?

Run `npm run design-system:dev` and confirm color palette

Question for @ItsJonQ @epiqueras (or anyone else)

When adding a component to Storybook, I've noticed issues around CSS, Button Group was one, see below for ColorPalette issue. Should these be fixed elsewhere, in the component itself, and not in the story? Should Storybook just reflect the state of the raw component?

For the Color Palette there are some problems with the Custom Color control (see screenshot)

Screenshot from 2019-10-17 06-39-53

### CODE REVIEW

**Comment by epiqueras** [comment]
The storybook is already loading all the styles that consumers get when importing `@wordpress/components`. This is just surfacing that the components are not very stand alone.

We should fix it at the component/package level and not at the storybook level, because consumers will have the same issues.

**Comment by mkaz** [comment]
@epiqueras makes sense, and matches with my assumption: Storybook should just be surfacing the components, and not attempt to style. It is just live documentation of what you get when using the component.

**Comment by youknowriad** [comment]
Aren't the styling issues here related to the global class `screen-reader-text` we rely on? I think this class can be refactored to be a component `<ScreenReaderText>text</ScreenReaderText>` and bundle the style in the component to fix the issue.

**Comment by mkaz** [comment]
I created #18022 to add the component. I also noticed the style.scss needs to be loaded by hand in stories, the components package bundles it all up. The #18022 PR also updates webpack.config.js to support and load style.scss in stories.

**Comment by epiqueras** [comment]
>Aren't the styling issues here related to the global class screen-reader-text we rely on?

Yeah, that means it's not stand alone.

>I also noticed the style.scss needs to be explictly imported in stories, the components package bundles it all up

`import '../build-style/style.css';` gets called in `packages/components/storybook/config.js`. Components don't import styles themselves so their stories shouldn't neither.

**Comment by mkaz** [comment]
> Components don't import styles themselves so their stories shouldn't neither.

@epiqueras - this is fine, if we add the ScreenReaderText component in #18022 it will add the `.screen-reader-text` to the bundled `import '../build-style/style.css'` so the rest of the component will benefit. 
I'll remove the style loader code from that PR.

What is the guidance for a user using the components? Just to include all the CSS. I guess this makes sense for the case of stylized components. Doesn't have to be decided here

**Comment by mkaz** [comment]
@epiqueras I removed the style loading from #18022 

I think there is a bug with our Storybook setup, it doesn't look like the `build-style/style.css` doesn't get rebuilt when running `npm run design-system:dev` - I needed to run `npm run designs-system:build` first to get it to pick up the CSS.

**Comment by epiqueras** [comment]
>What is the guidance for a user using the components? Just to include all the CSS.

So far, yes.

>I think there is a bug with our Storybook setup, it doesn't look like the build-style/style.css doesn't get rebuilt when running npm run design-system:dev - I needed to run npm run designs-system:build first to get it to pick up the CSS.

Looks like we should run `npm run build:packages` before starting to watch with `npm run dev:packages`.

**Comment by mkaz** [comment]
This PR is good to go. The changes that will improve the accessibility text showing are within the ColorPicker component PR #18013 that the ColorPalette uses. This PR is not dependent on that one, either can go at any time. This one will just benefit.

File: packages/components/src/color-palette/stories/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ColorPalette from '../';
+
+export default { title: 'Color Palette', component: ColorPalette };
```
Comment (epiqueras) [inline packages/components/src/color-palette/stories/index.js]:
```suggestion
export default { title: 'ColorPalette', component: ColorPalette };
```

File: packages/components/src/color-palette/stories/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+export const _default = () => {
+	const [ color, setColor ] = useState( '#f00' );
```
Comment (epiqueras) [inline packages/components/src/color-palette/stories/index.js]: We can't use hooks in stories. Storyshots fail. See #18030.

File: packages/components/src/color-palette/stories/index.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	return (
+		<ColorPalette
+			colors={ colors }
+			value={ color }
+			onChange={ setColor }
+		/>
+	);
+};
+
+export const WithKnobs = () => {
```
Comment (epiqueras) [inline packages/components/src/color-palette/stories/index.js]:
```suggestion
export const withKnobs = () => {
```

File: packages/components/src/color-palette/stories/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	const [ color, setColor ] = useState( '#f00' );
```
Comment (epiqueras) [inline packages/components/src/color-palette/stories/index.js]: Same here.

**Review by epiqueras — CHANGES_REQUESTED** [review]

File: packages/components/src/color-palette/stories/index.js (hunk 2 above)
Comment (mkaz) [reply packages/components/src/color-palette/stories/index.js]: Fixed.

File: packages/components/src/color-palette/stories/index.js (hunk 4 above)
Comment (mkaz) [reply packages/components/src/color-palette/stories/index.js]: Fixed.

File: packages/components/src/color-palette/stories/index.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ColorPalette from '../';
+
+export default { title: 'ColorPalette', component: ColorPalette };
+
+const ColorPaletteWithState = ( { ...props } ) => {
```
Comment (epiqueras) [inline packages/components/src/color-palette/stories/index.js]:
```suggestion
const ColorPaletteWithState = ( props ) => {
```

**Review by epiqueras — CHANGES_REQUESTED** [review]

**Review by epiqueras — APPROVED** [review]

**Pull request merged**